src/main.rs

```diff
@@ -440,6 +447,25 @@ fn print_add_response(hash: Hash, entries: Vec<ProvideResponseEntry>) {
 const PROGRESS_STYLE: &str =
     "{msg}\n{spinner:.green} [{elapsed_precise}] [{wide_bar:.cyan/blue}] {bytes}/{total_bytes} ({eta})";
 
+#[cfg(feature = "metrics")]
+fn init_metrics_collection(
+    metrics_addr: Option<SocketAddr>,
+) -> Option<tokio::task::JoinHandle<()>> {
+    init_metrics();
```

Comment:
This is still called unconditionally, IIUC this means even if `metrics_addr` is `None` it will track the metrics, but not start the server to export them.  Should this not be inside the `if let Some` just below so that all macros become no-ops?

Reply:
Same as an earlier comment. Currently there's only the endpoint to expose metrics, but we'll want other methods of dumping data soon and collection should be independent of serving.